### PR TITLE
Update PR template to produce PR descriptions that are easier to read

### LINKED ISSUE #758: Update PR template to produce easier-to-read PR descriptions
### Opportunities for improvement (IMO)

 - [ ] Developers tend to leave the template instructions in the final PR descriptions
 - [ ] Menu options in template do not account for all code in this repo
 - [ ] "Type of change" options are "3 apples and 1 orange"

### Before (old template, not filled in)

image

### After (new template, filled in)

image

### PR DESCRIPTION
**Pull request by eecavanna**
In this branch, I updated the PR template in ways I thought would make the resulting PR descriptions easier for people to consume.

### Details

The PR template previously began with instructions that I think many developers (including me) did not fully read each time they created a PR; and, as a result, those instructions (some of which say, effectively, "delete me after doing what I say") would remain in the final PR description. This decreased the signal-to-noise ratio for PR reviewers. Also it sometimes made it difficult for reviewers or passersby to determine—at a glance—whether the author actually _filled out_ the template to make a PR description, or just submitted _the template, itself_ as the PR description. This was my main motivation for overhauling the PR template.

In this branch, I took advantage of HTML comments `<!-- ... -->` to include instructions for the PR creator, while keeping those instructions hidden from the PR reader.

In addition, the "type of change" menu options in the original PR template seemed to me to be along three different "dimensions" (i.e. whether something was a bug fix versus a feature, whether something was a breaking change or not, and whether the change required an update to any documentation). In the PR template I've come up with here, I've tried to limit each menu to a single dimension.

### Related issue(s)

Fixes #758 

### Related subsystem(s)

- [ ] Runtime API (except the Minter)
- [ ] Minter
- [ ] Dagster
- [ ] Project documentation (in the `docs` directory)
- [ ] Mongo migrations
- [x] Other

### Testing

- [x] I tested these changes (explain below)
- [ ] I have not tested these changes yet

I tested these changes by filling out this PR template for this PR! ...which did result in me making changes to it.

### Documentation

- [ ] I **have not checked** for relevant documentation yet (e.g. in the `docs` directory)
- [x] I have **updated** all relevant documentation so it will remain accurate
- [ ] Other (explain below)

### Maintainability

- [x] Every new function I defined includes a docstring _(test functions are exempt from this)_
- [x] Every function parameter I introduced includes a type hint (e.g. `study_id: str, num_biosamples: int`)
- [x] All "todo" or "fixme" comments I added begin with either `# TODO` or `# FIXME`
- [ ] I used `black` to format all the files I added/modified
- [x] The PR title is in the imperative mood (e.g. "Do X") and not the declarative mood (e.g. "Does X" or "Did X")

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,37 +1,104 @@
...
+
+    Example: If you modified the `/studies/{study_id}` API endpoint,
+             mark the checkbox next to "Runtime API (except the Minter)".
+-->
+
+- [ ] Runtime API (except the Minter)
+- [ ] Minter
+- [ ] Dagster
+- [ ] Project documentation (in the `docs` directory)
+- [ ] Mongo migrations
```
Comment (eecavanna) [inline .github/pull_request_template.md:57]:
```suggestion
- [ ] MongoDB migrations
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,37 +1,104 @@
...
+### Maintainability
+
+<!--
+    7. Maintainability (required)
+
+    Indicate whether you have done each of these things that can make code easier to maintain,
+    whether by others or by your future self.
+-->
+
+- [ ] Every new function I defined includes a docstring _(test functions are exempt from this)_
```
Comment (eecavanna) [inline .github/pull_request_template.md:101]:
```suggestion
- [ ] Every Python function I defined includes a docstring _(test functions are exempt from this)_
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,37 +1,104 @@
...
+- [ ] Every function parameter I introduced includes a type hint (e.g. `study_id: str, num_biosamples: int`)
```
Comment (eecavanna) [inline .github/pull_request_template.md:102]:
```suggestion
- [ ] Every Python function parameter I introduced includes a type hint (e.g. `study_id: str`)
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,37 +1,104 @@
...
+- [ ] All "todo" or "fixme" comments I added begin with either `# TODO` or `# FIXME`
```
Comment (eecavanna) [inline .github/pull_request_template.md:103]:
```suggestion
- [ ] All "to do" or "fix me" Python comments I added begin with either `# TODO` or `# FIXME`
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,37 +1,104 @@
...
+- [ ] I used `black` to format all the files I added/modified
```
Comment (eecavanna) [inline .github/pull_request_template.md:104]:
```suggestion
- [ ] I used `black` to format all the Python files I created/modified
```

**Comment by eecavanna** [comment]
I am curious how my teammates feel about filling this template out and how they feel about the result it produces (e.g. in this PR).

**Review by pkalita-lbl — DISMISSED** [review]
I think this is a great way to make things less boilerplate-y

**Comment by eecavanna** [comment]
Thanks! I agree. I'll wait until one more person has taken a look (since today was a holiday for some people) before merging.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -1,37 +1,108 @@
...
+    │ may involve attaching example files or ad hoc test instructions.        │
+    │                                                                         │
+    │ Example: "I tested these changes by adding a pytest test that ensures   │
+    │           the database does not contain a Study whose ID is `foo`,      │
+    │           then submits an HTTP request to `/studies/foo` and confirms   │
+    │           the response status is 404."                                  │
+    └─────────────────────────────────────────────────────────────────────────┘-->
 
+- [ ] I tested these changes (explain below)
+- [ ] I have not tested these changes yet
```
Comment (eecavanna) [inline .github/pull_request_template.md:77]:
```suggestion
- [ ] I did not test these changes
```

**Review by aclum — APPROVED** [review]

**Pull request merged**